artiq/coredevice/urukul.py

```diff
@@ -343,13 +365,41 @@ def get_att_mu(self):
         self.bus.write(0)  # shift in zeros, shift out current value
         self.bus.set_config_mu(SPI_CONFIG | spi.SPI_END, 32,
                                SPIT_ATT_WR, CS_ATT)
-        delay(10*us)
+        delay(10 * us)
         self.att_reg = self.bus.read()
         self.bus.write(self.att_reg)  # shift in current value again and latch
         return self.att_reg
 
     @kernel
-    def set_sync_div(self, div):
+    def get_att_mu_(self, channel: TInt32) -> TInt32:
+        """Get digital step attenuator value for a channel in machine units.
+
+        The result is stored and will be used in future calls of
+        :meth:`set_att_mu` and :meth:`set_att`.
+
+        .. seealso:: :meth:`get_att_mu`
+
+        :param channel: Attenuator channel (0-3).
+        :return: 8-bit digital attenuation setting:
+            255 minimum attenuation, 0 maximum attenuation (31.5 dB)
+        """
+        return int32((self.get_att_mu() >> (channel * 8)) & 0xff)
+
+    @kernel
+    def get_att(self, channel: TInt32) -> TFloat:
```

Comment:
For consistency, would it make sense to make this return all the attenuations in dB and add ``get_channel_att()``? The symmetry between machine and human units wouldn't be perfect (I guess that return the 32 bits of the SR in ``get_att_mu()`` is more appropriate than a tuple) but IMHO better.

Reply:
I renamed the function. `get_att()` can not be implemented without storing extra information because a locally generated list can not be returned from a function (i.e. object lifetime). To implement this, there needs to be a second `att_reg` that stores the float values. That sounds a bit overkill to me and I think we can leave that function out. Do you agree?

Reply:
Thanks! I agree: I guess that most code will use ``get_channel_att`` anyway.